tests/test_logger.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (C) 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import datetime
+import logging as deflogging  # Default logging
+
+import pytest
+
+from ansys.additive.core import LOG  # Global logger
+import ansys.additive.core.progress_logger as logger
+
+LOG_LEVELS = {"CRITICAL": 50, "ERROR": 40, "WARNING": 30, "INFO": 20, "DEBUG": 10}
+
+
+def test_global_logger_exist():
+    """Test for checking the accurate naming of the general Logger instance."""
+
+    assert isinstance(LOG.logger, deflogging.Logger)
+    assert LOG.logger.name == "PyAdditive_global"
+
+
+def test_global_logger_has_handlers():
+    """Test for checking that the general Logger has file_handlers and sdtout
+    file_handlers implemented."""
+
+    assert hasattr(LOG, "file_handler")
+    assert hasattr(LOG, "std_out_handler")
+    assert LOG.logger.hasHandlers
+    assert LOG.file_handler or LOG.std_out_handler
+
+
+def test_global_logger_logging(caplog: pytest.LogCaptureFixture):
+    """Testing the global PyAdditive logger capabilities. Forcing minimum logging level to
+    Debug, adding a message with different logging levels, checking the output and
+    restoring to original level.
+
+    Parameters
+    ----------
+    caplog : pytest.LogCaptureFixture
+        Fixture for capturing logs.
+    """
+    LOG.logger.setLevel("DEBUG")
```

Comment:
We should first set a local variable to the current log level rather than assuming it will always be `ERROR`. I could see us changing the default to `INFO` one day. This test would pass but set the level incorrectly at the end. It's unlikely the developer who changes the default would know to update this test. Another option is to create a local logger, `testlogger = Logger(level=logging.DEBUG, to_file=False, to_stdout=True)`, and use it instead of `LOG`.

Reply:
Using test setup and teardown to set log level back to defautl